### ISSUE TITLE
Mac M1 chntpw not building (Openssl 1.0 requirement not satisfied)

### ISSUE THREAD

**Issue by ashtonvalasko** [uup-dump/converter#19]
MacBook Air 2020 M1 Chip, Monterey 12.1 beta 3 (21C5039b).

When entering the command
**$ brew install sidneys/homebrew/chntpw**
This is the full text from terminal including the error messages:

-------------------------------
bash-3.2$ **brew install sidneys/homebrew/chntpw**
==> Downloading https://ftp.openssl.org/source/old/1.0.2/openssl-1.0.2u.tar.gz
Already downloaded: /Users/ash/Library/Caches/Homebrew/downloads/6a06bfa556bd474c7e4efb2eab11c7d43d4238b508bd91e74c438653e24b8a45--openssl-1.0.2u.tar.gz
==> Downloading https://github.com/sidneys/chntpw/archive/0.99.6.tar.gz
Already downloaded: /Users/ash/Library/Caches/Homebrew/downloads/dee692592b6fc0d6db43012210cf70e9ef7a6018e20818b5f15187cbd3a06e39--chntpw-0.99.6.tar.gz
==> Installing chntpw from sidneys/homebrew
==> Installing dependencies for sidneys/homebrew/chntpw: sidneys/homebrew/openssl@1.0
==> Installing sidneys/homebrew/chntpw dependency: sidneys/homebrew/openssl
==> perl ./Configure --prefix=/opt/homebrew/Cellar/openssl@1.0/1.0.2u --openssld
==> make
Last 15 lines from /Users/ash/Library/Logs/Homebrew/openssl@1.0/02.make:
 ^
x86_64cpuid.s:273:10: error: unknown token in expression
 cmpq $0,%rax
         ^
x86_64cpuid.s:273:10: error: invalid operand
 cmpq $0,%rax
         ^
x86_64cpuid.s:274:9: error: unknown token in expression
 cmoveq %rcx,%rax
        ^
x86_64cpuid.s:274:9: error: invalid operand
 cmoveq %rcx,%rax
        ^
make[1]: *** [x86_64cpuid.o] Error 1
make: *** [build_crypto] Error 1

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/sidneys/homebrew-homebrew/issues

These open issues may also help:
Mac M1 chntpw not building (Openssl 1.0 requirement not satisfied) https://github.com/sidneys/homebrew-homebrew/issues/2
------------------------------

I saw the error about Openssl, so I  installed it through Brew no problem, but then it's saying it's still not satisfied the prerequisite. I'm assuming there is some incompatibility with the M1 Chip. I've included a screenshot as well. If there's a loophole somewhere, let me know. I am submitting this error as it tells you to do when running the command. Thank you!

<img width="599" alt="Screen Shot 2021-11-28 at 21 24 14" src="https://user-images.githubusercontent.com/32396947/143799097-5f91d96d-6cb7-4f65-8999-c42c2ee8f68d.png">

**Comment by ashtonvalasko** [comment]
Update. This fixed it:

Easy Installation
curl -LO https://gist.github.com/minacle/e9dedb8c17025a23a453f8f30eced3da/raw/908b944b3fe2e9f348fbe8b8800daebd87b5966c/openssl@1.0.rb
curl -LO https://gist.github.com/minacle/e9dedb8c17025a23a453f8f30eced3da/raw/908b944b3fe2e9f348fbe8b8800daebd87b5966c/chntpw.rb
brew install --formula --build-from-source ./openssl@1.0.rb
brew install --formula --build-from-source ./chntpw.rb
rm ./openssl@1.0.rb ./chntpw.rb

YOU NEED TO $brew install aria2
YOU NEED TO CHMOD +X uup_download_macos
RUN ./uup_download_macos

BAM it starts downloading.

**Comment by manishie** [comment]
Thank you!  Worked like a charm.  :-)

**Comment by servin** [comment]
can't make it work on a MBP m1pro 

Error: chntpw: SHA256 mismatch

I've removed the file but still stuck there u.u

**Comment by Naveen7300** [comment]
Hii. I am beginner so i didn't get exactly what commands have to be written for this to work. I tried to do it myself but I am not getting anywhere with it.

**Comment by vincio71** [comment]
> > Update. This fixed it:
> > Easy Installation curl -LO https://gist.github.com/minacle/e9dedb8c17025a23a453f8f30eced3da/raw/908b944b3fe2e9f348fbe8b8800daebd87b5966c/openssl@1.0.rb curl -LO https://gist.github.com/minacle/e9dedb8c17025a23a453f8f30eced3da/raw/908b944b3fe2e9f348fbe8b8800daebd87b5966c/chntpw.rb brew install --formula --build-from-source ./[openssl@1.0.rb](mailto:openssl@1.0.rb) brew install --formula --build-from-source ./chntpw.rb rm ./[openssl@1.0.rb](mailto:openssl@1.0.rb) ./chntpw.rb
> > YOU NEED TO $brew install aria2 YOU NEED TO CHMOD +X uup_download_macos RUN ./uup_download_macos
> > BAM it starts downloading.
> 
> even this doesnt work for me, i got the following error(s?): brew install --formula --build-from-source ./[openssl@1.0.rb](mailto:openssl@1.0.rb)
> 
> ==> Fetching openssl@1.0 ==> Downloading https://gist.githubusercontent.com/felixbuenemann/5f4dcb30ebb3b86e1302e2ec305bac89/raw/b339a33ff072c9747df2 Already downloaded: /Users/auhustgammel/Library/Caches/Homebrew/downloads/db281f2b8a51c8a97d6bced562e99959b2c8b3241129895f8850506ccd1f8899--openssl-1.0.2u-darwin-arm64.patch ==> Downloading https://ftp.openssl.org/source/old/1.0.2/openssl-1.0.2u.tar.gz Already downloaded: /Users/auhustgammel/Library/Caches/Homebrew/downloads/6a06bfa556bd474c7e4efb2eab11c7d43d4238b508bd91e74c438653e24b8a45--openssl-1.0.2u.tar.gz ==> Downloading https://formulae.brew.sh/api/formula.jws.json #=#=- # # Warning: openssl@1.0 1.0.2u_1 is available and more recent than version 1.0.2u. ==> Patching ==> Applying openssl-1.0.2u-darwin-arm64.patch patching file Configure ==> perl ./Configure --prefix=/opt/homebrew/Cellar/openssl@1.0/1.0.2u --openssldir=/opt/homebrew/etc/openssl@1.0 no-ssl2 no ==> make ==> make test Last 15 lines from /Users/auhustgammel/Library/Logs/Homebrew/openssl@1.0/03.make: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:100B906B:elliptic curve routines:EC_POINT_set_affine_coordinates_GF2m:point is not on curve:ec_lib.c:994: 8058625004:error:1007C06B:elliptic curve routines:EC_POINT_set_affine_coordinates_GFp:point is not on curve:ec_lib.c:968: 8058625004:error:1007C06B:elliptic curve routines:EC_POINT_set_affine_coordinates_GFp:point is not on curve:ec_lib.c:968: 8058625004:error:1007C06B:elliptic curve routines:EC_POINT_set_affine_coordinates_GFp:point is not on curve:ec_lib.c:968: make[1]: *** [test_ec] Error 1 make: *** [tests] Error 2
> 
> Do not report this issue to Homebrew/brew or Homebrew/homebrew-core!

Did you find some solution?

**Comment by Nekroph** [comment]
This worked for me.
```
brew tap minacle/chntpw
brew install minacle/chntpw/chntpw

```
